File: src/models/Note.js
const mongoose=require('mongoose');

const noteSchema=mongoose.Schema({
    id:{
        type:String,
        unique:true,
        require:true,    
    },
    userId:{
        type:String,
        require:true,    
    },
    title:{
        type:String,
        require:true,    
    },
    content:{
        type:String,
           
    },
    dateAdded:{
        type:Date,
        default:Date.now,    
    }
});

module.exports= mongoose.model("Note",noteSchema);

File: lib/Models/Note.dart
class Note {
  String? id;
  String? userId;
  String? title;
  String? content;
  DateTime? dateTime;

  Note({this.id, this.userId, this.title, this.content, this.dateTime});

  factory Note.fromMap(Map<String, dynamic> map) {
    return Note(
      id: map["id"],
      userId: map["userId"],
      title: map["title"],
      content: map["content"],
      dateTime: DateTime.tryParse(map["dateAdded"]),
    );
  }

  Map<String, dynamic> toMap() {
    return {
      "id": id,
      "userId": userId,
      "title": title,
      "content": content,
      "dateAdded": dateTime!.toIso8601String(),
    };
  }
}
